### DISCUSSION TITLE
Question: about originalPdfDocument.get_page_images(pageNumber, full=True), can't get full images list.

### DISCUSSION THREAD

**Discussion by AwesomeYuer** [pymupdf/PyMuPDF#2759] [Q&A] [answered]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
I want get full `image list` with parameter `full=true` , even though some images are same in pdf.
I can't get all image items from below code.
There are 5 images in the pdf, 4 of 5 are same,
So only top 2 unique image item are output.

Would you help to resolve it?

Thanks in advance.

## To Reproduce (mandatory)

1. the Sample problem PDF document as below

https://github.com/AwesomeYuer/public-misc-share/blob/main/Simple.Rowspan.for.Pdf.Print.pdf

the pdf document is print by one html as below, all the images are inline base64 html format

https://github.com/AwesomeYuer/public-misc-share/blob/main/Simple.Rowspan.for.Pdf.Print.html

2.run the code in `ipynb` in `vscode`

```python

pdfDocument = fitz.open(path)
page = pdfDocument[0] # Page 2
page.wrap_contents()
images = pdfDocument.get_page_images(0, full=True)
for i, item in enumerate(images):
    print(i, item)

```

## Expected behavior (optional)
Describe what you expected to happen (if not obvious).

I want get full `image list` with parameter `full=true` , even though some images are same in pdf.

The full output list should be 5 items

## Screenshots (optional)

![image](https://github.com/pymupdf/PyMuPDF/assets/1026479/28d2f353-ed92-4017-97f1-94f3cd1f8273)

![image](https://github.com/pymupdf/PyMuPDF/assets/1026479/77d5d383-e26e-44f2-8878-8d597f0633da)

## Your configuration (mandatory)
 - Operating system, potentially version and bitness
 - Python version, bitness
 - PyMuPDF version, installation method (**wheel** or **generated** from source).

For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

```
3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0] 
 linux 
 
PyMuPDF 1.23.5: Python bindings for the MuPDF 1.23.4 library.
Version date: 2023-10-11 00:00:01.
Built for Python 3.11 on linux (64-bit).
```

**Comment by JorjMcKie** [comment]
This is a typical "Discussions" post - transferring to the respective tab.

**Comment by JorjMcKie** [comment, marked as answer]
Method `page.get_images()` only shows which image references are contained in the page's PDF object definition. This may or may not be the same set of images that the page actually displays. Whether you use the `full` parameter or not: this has a completely different meaning.
Please consult the documentation for more background.

To get a list of all image **_display_** **_commands_** that the page actually issues, use `page.get_image_info()` (please consult documentation for parameter details). This also contains duplicate images and images that may not have an xref (inline images).
Also, the sequence in that list is the sequence in which the display commands are being executed.

**Reply by AwesomeYuer** [reply]
@JorjMcKie thank you very much.

@AwesomeYuer 

Would you help to modify the below code to get `xref` by using `page.get_image_info`?

```python

pdfDocument = fitz.open(path)
page = pdfDocument[0]
page.wrap_contents()
images = pdfDocument.get_page_images(0, full=True)

for i, item in enumerate(images):
    print(i, item, type(item))
    old_xref = item[0]


print("===================================")


imagesInfo = page.get_image_info()

rects_list = []

for i, item in enumerate(imagesInfo):
    print(i, item, type(item))
    
    bb = item["bbox"]
    text_rect = fitz.Rect(bb[0], bb[1], bb[2], bb[3])
    rects_list.append(text_rect)
    
    # look here
    # Would you help to get the 'xref' of image from `image dict`?
    # ??????????
    old_xref = item ??????????????????
    # thanks in advance
```

**Reply by AwesomeYuer** [reply]
I know set parameter `xrefs=True`

**Reply by JorjMcKie** [reply]
Ok - so you are good?

Look, instead of doing this:
```python
    bb = item["bbox"]
    text_rect = fitz.Rect(bb[0], bb[1], bb[2], bb[3])
```
... simply do this:
```python
    text_rect = fitz.Rect(item["bbox"])
```
`fitz.Rect` understands any numeric sequence of length 4 and can directly make a rectangle from it.

**Reply by AwesomeYuer** [reply]
@JorjMcKie 

I am good!

Big thanks for your answer and suggestion in time!
